quixstreams/models/topics/admin.py

```diff
@@ -0,0 +1,194 @@
+
+        :param futures: a dict of futures as generated by Confluent's
+            `AdminClient.create_topics()`
+        :param timeout: How long to confirm topics before raising exception
+        """
+        exceptions = {}
+        successful = []
+        existed = []
+        stop_time = time.time() + timeout
+        while futures and time.time() < stop_time:
+            time.sleep(1)
+            for topic_name in list(futures.keys()):
+                future = futures[topic_name]
+                if future.done():
+                    try:
+                        future.result()
+                        successful.append(topic_name)
+                    except KafkaException as e:
+                        # Topic was maybe created by another instance
+                        if e.args[0].name() == "TOPIC_ALREADY_EXISTS":
+                            existed.append(topic_name)
+                        else:
+                            exceptions[topic_name] = e.args[0].str()
+                    # Not sure how these get raised, but they are supposedly possible
+                    except (TypeError, ValueError) as e:
+                        exceptions[topic_name] = e
+                    del futures[topic_name]
+        if successful:
+            logger.info(f"Successfully created topics: {successful}")
+        if existed:
+            logger.info(f"These topics already exist: {existed}")
+        if exceptions:
+            raise CreateTopicFailure(exceptions)
+        if futures:
+            raise CreateTopicTimeout(
+                f"Timed out waiting for creation status for topics:\n"
+                f"{pprint.pformat([topic_name for topic_name in futures])}"
+            )
+
+    def create_topics(
+        self, topics: List[Topic], timeout: int = 10, finalize_timeout: int = 20
+    ):
+        """
+        Create the given list of topics and confirm they are ready.
+
+        Also raises an exception with detailed printout should the creation
+        fail (it ignores issues for a topic already existing).
+
+        :param topics: a list of `Topic`
+        :param timeout: timeout of the creation broker request
+        :param finalize_timeout: the timeout of the topic finalizing ("ready")
+        """
+        existing = self.list_topics().keys()
+        if not (topics := [topic for topic in topics if topic.name not in existing]):
+            logger.info("All topics already exist!")
+            return
+        try:
+            self._finalize_create(
+                self._admin_client.create_topics(
+                    convert_topic_list(topics),
+                    request_timeout=timeout,
+                ),
+                finalize_timeout,
+            )
+        except CreateTopicFailure as e:
+            failure = {
+                topic.name: {
+                    "failure_reason": e.args[0][topic.name],
+                    "topic_config": topic.config.__dict__,
```

Comment:
Please don't use `__dict__` like this. 
There's a `dataclasses.asdict()` method for it.